plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx

```diff
@@ -117,13 +111,15 @@ export const MembersListCard = (_props: {
 }) => {
   const { entity: groupEntity } = useEntity<GroupEntity>();
   const {
-    metadata: { name: groupName },
+    metadata: { name: groupName, namespace: grpNamespace },
     spec: { profile },
   } = groupEntity;
   const catalogApi = useApi(catalogApiRef);
 
   const displayName = profile?.displayName ?? groupName;
 
+  const groupNamespace = grpNamespace || ENTITY_DEFAULT_NAMESPACE;
+
   const { loading, error, value: members } = useAsync(async () => {
     const membersList = await catalogApi.getEntities({
```

Comment:
This whole filtering logic could be replaced by [`useRelatedEntities` from `plugin-catalog-react`](https://github.com/backstage/backstage/blob/c36a7847e49a1b61f76cff4451d30461c96e1246/plugins/catalog-react/src/hooks/useRelatedEntities.ts):

```ts
const groupMembersList = useRelatedEntities(groupEntity, {type: RELATION_MEMBER_OF, kind: 'User'});
```

But I'm not sure if the implementation is more efficient (getting all `User`s and filter them vs. getting all relations and query all users individually). I guess it depends on how many users are in an average sized group. @freben thoughts?

Reply:
We really need https://github.com/backstage/backstage/issues/4499 ...

Yeah use this code for now. I see the hook doesn't limit the issue concurrency, and before adopting it more maybe we should make the code issue batch queries via the getEntities method instead. (Giving multiple values to `filter`, one for each kind:namespace/name triplet).

Reply:
Another thing we could do to lighten the load is to use the `fields` parameter to the api, to only return the exact entity fields that we need to render the grid items.

Reply:
But this is good to go as far as I can tell!